tools/buildsys/src/manifest.rs

```diff
@@ -167,100 +175,111 @@ kernel-parameters = [
     }
 
     /// Convenience method to return the list of source groups.
-    pub(crate) fn source_groups(&self) -> Option<&Vec<PathBuf>> {
+    pub fn source_groups(&self) -> Option<&Vec<PathBuf>> {
         self.build_package().and_then(|b| b.source_groups.as_ref())
     }
 
     /// Convenience method to return the list of external files.
-    pub(crate) fn external_files(&self) -> Option<&Vec<ExternalFile>> {
+    pub fn external_files(&self) -> Option<&Vec<ExternalFile>> {
         self.build_package().and_then(|b| b.external_files.as_ref())
     }
 
     /// Convenience method to return the package name override, if any.
-    pub(crate) fn package_name(&self) -> Option<&String> {
+    pub fn package_name(&self) -> Option<&String> {
         self.build_package().and_then(|b| b.package_name.as_ref())
     }
 
     /// Convenience method to find whether the package is sensitive to variant changes.
-    pub(crate) fn variant_sensitive(&self) -> Option<&VariantSensitivity> {
+    pub fn variant_sensitive(&self) -> Option<&VariantSensitivity> {
         self.build_package()
             .and_then(|b| b.variant_sensitive.as_ref())
     }
 
     /// Convenience method to return the list of included packages.
-    pub(crate) fn included_packages(&self) -> Option<&Vec<String>> {
+    pub fn included_packages(&self) -> Option<&Vec<String>> {
         self.build_variant()
             .and_then(|b| b.included_packages.as_ref())
     }
 
     /// Convenience method to return the image format override, if any.
-    pub(crate) fn image_format(&self) -> Option<&ImageFormat> {
+    pub fn image_format(&self) -> Option<&ImageFormat> {
         self.build_variant().and_then(|b| b.image_format.as_ref())
     }
 
     /// Convenience method to return the image layout, if specified.
-    pub(crate) fn image_layout(&self) -> Option<&ImageLayout> {
-        self.build_variant().and_then(|b| b.image_layout.as_ref())
+    pub fn image_layout(&self) -> Option<&ImageLayout> {
+        self.build_variant().map(|b| &b.image_layout)
     }
 
     /// Convenience method to return the supported architectures for this variant.
-    pub(crate) fn supported_arches(&self) -> Option<&HashSet<SupportedArch>> {
+    pub fn supported_arches(&self) -> Option<&HashSet<SupportedArch>> {
         self.build_variant()
             .and_then(|b| b.supported_arches.as_ref())
     }
 
     /// Convenience method to return the kernel parameters for this variant.
-    pub(crate) fn kernel_parameters(&self) -> Option<&Vec<String>> {
+    pub fn kernel_parameters(&self) -> Option<&Vec<String>> {
         self.build_variant()
             .and_then(|b| b.kernel_parameters.as_ref())
     }
 
-    /// Convenience method to return the GRUB features for this variant.
-    pub(crate) fn grub_features(&self) -> Option<&Vec<GrubFeature>> {
-        self.build_variant().and_then(|b| b.grub_features.as_ref())
+    /// Convenience method to return the enabled image features for this variant.
```

Comment:
Are image features only ever going to be boolean flags? Should you allow for the future possibility than an image feature can have a string or numeric value?